### ISSUE TITLE
Warnings that need to get suppressed

### ISSUE THREAD

**Issue by dlakaplan** [nanograv/PINT#1224]
Please list here any warning messages you see that you don't want to see (info about what creates them might help, but isn't crucial).  I can then add them to the filter.

**Comment by dlakaplan** [comment]
`ERFA function "dtf2d" yielded 1 of "dubious year (Note 6)"`

**Comment by paulray** [comment]
```
The datapath rcparam was deprecated in Matplotlib 3.2.1 and will be removed two minor releases later.
WARNING  (pint.logging                  ): None 
The verbose.level rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The verbose.fileo rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The text.latex.unicode rcparam was deprecated in Matplotlib 3.0 and will be removed in 3.2.
WARNING  (pint.logging                  ): None 
The savefig.frameon rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None Setting the datapath via matplotlibrc is deprecated 3.2 and will be removed in in 3.3
. 
WARNING  (pint.logging                  ): None 
The datapath rcparam was deprecated in Matplotlib 3.2.1 and will be removed two minor releases later.
WARNING  (pint.logging                  ): None 
The datapath rcparam was deprecated in Matplotlib 3.2.1 and will be removed two minor releases later.
WARNING  (pint.logging                  ): None 
The savefig.frameon rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The text.latex.unicode rcparam was deprecated in Matplotlib 3.0 and will be removed in 3.2.
WARNING  (pint.logging                  ): None 
The verbose.fileo rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The verbose.level rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The datapath rcparam was deprecated in Matplotlib 3.2.1 and will be removed two minor releases later.
WARNING  (pint.logging                  ): None 
The verbose.level rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The verbose.fileo rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
WARNING  (pint.logging                  ): None 
The text.latex.unicode rcparam was deprecated in Matplotlib 3.0 and will be removed in 3.2.
WARNING  (pint.logging                  ): None 
The savefig.frameon rcparam was deprecated in Matplotlib 3.1 and will be removed in 3.3.
```

**Comment by dlakaplan** [comment]
`The refresh function was deprecated in Matplotlib 3.3 and will be removed two minor releases later.` (from @swiggumj )

**Comment by dlakaplan** [comment]
`<class 'erfa.core.ErfaWarning'> ERFA function "pmsafe" yielded 1 of "distance overridden (Note 6)"`

**Comment by dlakaplan** [comment]
> ERFA function "dtf2d" yielded 1 of "dubious year (Note 6)"

is related to https://github.com/nanograv/PINT/blob/49f0385efe7fe87047e59267cc3e181fc95bc92a/src/pint/observatory/clock_file.py (thanks @rossjjennings )

**Comment by dlakaplan** [comment]
Question: should all of the ERFA warnings be suppressed?  Or just shown once?  It's easy to suppress them, but I'm not sure if they are useful.  Could just do one at a time.  Showing once it slightly harder since in general the practice had been to only apply that filter to INFO or below, but I can change that if needed.

**Comment by scottransom** [comment]
@paulray Are you using a custom rcparams file?  Or some .matplotlibrc file?  I wonder if that is where those warnings are coming from?

**Comment by dlakaplan** [comment]
I've seen different sorts of matplotlib messages on different systems.  Those I know how to fix immediately since there's no question what to do about them.

**Comment by paulray** [comment]
Those rcparams warnings came from a run by Mike Wolff that he was asking me about. I don't think he has a custom rcparams, so I'm not sure what causes them.  Basically they are just an example of the tons of those kind of deprecations warnings that pop up.

**Comment by paulray** [comment]
When I save a plot as PDF in pintk I get these warnings:
```
WARNING  (pint.logging                  ): None Glyph 0 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 1 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 2 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 3 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 4 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 5 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 6 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 7 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 8 ) missing from current font.
WARNING  (pint.logging                  ): None Glyph 9 (	) missing from current font.
WARNING  (pint.logging                  ): None Glyph 10 (
) missing from current font.
WARNING  (pint.logging                  ): None Glyph 11 (
                                                          ) missing from current font.
WARNING  (pint.logging                  ): None Glyph 12 (
                                                          ) missing from current font.
) missing from current font.            ): None Glyph 13 (
WARNING  (pint.logging                  ): None Glyph 14 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 15 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 16 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 17 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 18 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 19 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 20 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 21 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 22 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 23 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 24 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 25 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 26 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 27 (missing from current font.
WARNING  (pint.logging                  ): None Glyph 28 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 29 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 30 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 31 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 127 () missing from current font.
WARNING  (pint.logging                  ): None Glyph 65534 (\ufffe) missing from current font.
WARNING  (pint.logging                  ): None Glyph 65534 (\ufffe) missing from current font.
WARNING  (pint.logging                  ): None Glyph 65534 (\ufffe) missing from current font.
WARNING  (pint.logging                  ): None Glyph 65534 (\ufffe) missing from current font.
WARNING  (pint.logging                  ): None Glyph 65534 (\ufffe) missing from current font.
```

**Comment by paulray** [comment]
> I would vote no on suppressing all ERFA warnings. They're often useful indications that something problematic is happening, which is why I spent time trying to track down this one.

If we don't suppress them all, we should make sure that we don't generate them from places we know are not an issue, like clock files with late end dates for extrapolation. Otherwise, people just learn to ignore them since they come up all the time, and they end up having no value.

**Comment by dlakaplan** [comment]
Those `dtf2d` warnings were explicitly supposed to be ignore, but the logging mods overrode that.  I am putting that back.

I'm still trying to track down the `pmsafe` warnings.  I generically know the origin of them but don't know the exact line.

**Comment by dlakaplan** [comment]
I don't think I've seen those missing glyph warnings.  Are those printing directly from a script or from `pintk`?  

My worry is that if we try to get to all warnings related to individual setups then we will not have a good solution for all, which is why I was considering global config files (https://github.com/nanograv/PINT/issues/112).

**Comment by paulray** [comment]
> I don't think I've seen those missing glyph warnings. Are those printing directly from a script or from `pintk`?

That was clicking the save button in `pintk` and selecting PDF output. I don't get them if I select PNG output.

**Comment by rossjjennings** [comment]
How about `DDK model uses KIN as inclination angle. SINI will not be used. This happens every time a DDK model is constructed.`? It's awkward to have a warning that happens every time a DDK model is constructed. Maybe it could be downgraded to INFO level? It seems to be happening because something tries to set SINI during the construction of a DDK model, even when there's no SINI line in the par file. The code that raises the warning is here: https://github.com/nanograv/PINT/blob/49f0385efe7fe87047e59267cc3e181fc95bc92a/src/pint/models/stand_alone_psr_binaries/DDK_model.py#L107

**Comment by dlakaplan** [comment]
Addressed by #1232
